pkg/rpc/request/types_test.go

```diff
@@ -0,0 +1,49 @@
+package request
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"testing"
+)
+
+type readCloser struct {
+	io.Reader
+}
+
+func (readCloser) Close() error { return nil }
+
+func BenchmarkUnmarshal(b *testing.B) {
```

Comment:
I'm not sure, do we need this benchmark? It's kind of incomplete in terms of non-error requests processing because further parameters unmarshalling is performed by request handlers also.